pkg/event/event.go

```diff
@@ -0,0 +1,86 @@
+package event
+
+import (
+	"context"
+	"sync"
+
+	"github.com/openshift-online/maestro/pkg/api"
+)
+
+// EventClient is a client that can receive resource status change events.
+type EventClient struct {
+	clusterName string
+	resChan     chan *api.Resource
+}
+
+// NewEventClient creates a new event client.
+func NewEventClient(clusterName string) *EventClient {
+	return &EventClient{
+		clusterName: clusterName,
+		resChan:     make(chan *api.Resource),
+	}
+}
+
+// Receive returns a channel that can be used to receive resource status change events.
+func (c *EventClient) Receive() <-chan *api.Resource {
+	return c.resChan
+}
+
+// EventHub is a hub that can broadcast resource status change events to registered clients.
+type EventHub struct {
+	mu sync.RWMutex
+
+	// Registered clients.
+	clients map[*EventClient]struct{}
+
+	// Inbound messages from the clients.
+	broadcast chan *api.Resource
+}
+
+// NewEventHub creates a new event hub.
+func NewEventHub() *EventHub {
+	return &EventHub{
+		clients:   make(map[*EventClient]struct{}),
```

Comment:
using EventClient as a key seems a bit dangerous if we add more thing in the event client. Can we just use clusterName or a internal ID?

Reply:
This is intent.
if we have two different status subscribers(event client): client1 and client2.
they both subscribe resource status from cluster `cluster1` at the same time, then the event hub should send the resource status to client1 and client2.
If we use cluster name as key, the resource status will only be sent to client1 or client2.

Does it make sense?